sdk/keyvault/keyvault-admin/test/public/backupClient.spec.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "chai";
+import { env, Recorder } from "@azure/test-utils-recorder";
+import { KeyClient } from "@azure/keyvault-keys";
+
+import { KeyVaultBackupClient } from "../../src";
+import { authenticate } from "../utils/authentication";
+import { testPollerProperties } from "../utils/recorder";
+import { formatName, getFolderName } from "../utils/common";
+
+describe("KeyVaultBackupClient", () => {
+  let client: KeyVaultBackupClient;
+  let keyClient: KeyClient;
+  let recorder: Recorder;
+
+  beforeEach(async function() {
+    const authentication = await authenticate(this);
+    client = authentication.backupClient;
+    keyClient = authentication.keyClient;
+    recorder = authentication.recorder;
+  });
+
+  afterEach(async function() {
+    await recorder.stop();
+  });
+
+  // The tests follow
+
+  it("beginBackup", async function() {
+    const blobStorageUri = env.BLOB_STORAGE_URI;
+    const sasToken = env.BLOB_STORAGE_SAS_TOKEN;
+    const backupPoller = await client.beginBackup(blobStorageUri, sasToken, testPollerProperties);
+    const backupURI = await backupPoller.pollUntilDone();
+    assert.ok(!!backupURI.match(blobStorageUri));
+  });
+
+  it("beginBackup, then beginRestore", async function() {
+    const blobStorageUri = env.BLOB_STORAGE_URI;
+    const sasToken = env.BLOB_STORAGE_SAS_TOKEN;
+    const backupPoller = await client.beginBackup(blobStorageUri, sasToken, testPollerProperties);
+    const backupURI = await backupPoller.pollUntilDone();
+    assert.ok(!!backupURI.match(blobStorageUri));
+
+    const folderName = getFolderName(blobStorageUri);
+    const restorePoller = await client.beginRestore(
+      blobStorageUri,
+      sasToken,
+      folderName,
+      testPollerProperties
+    );
+    await restorePoller.pollUntilDone();
```

Comment:
Do we want any verification for restore result?

Reply:
It makes sense to me to check that the poller reported no errors. If the suggestion is to verify that the operation did what it was supposed to do, I'd rather avoid doing that setup today, plus that would be more a integration test, since it's about whether the service ended up doing what it reported to have done, if that makes sense.